src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -72,34 +231,33 @@ const SKUpdate& SKNMEA2000Parser::parse128259(const SKSourceInput& input, const
     if (!N2kIsNA(waterSpeed)) {
       update->setNavigationSpeedThroughWater(waterSpeed);
     }
-    if (!N2kIsNA(groundSpeed)) {
-      update->setNavigationSpeedOverGround(groundSpeed);
-    }
 
     _sku = update;
     return *_sku;
   }
   else {
-    INFO("Unable to parse N2kMsg with PGN %i", msg.PGN);
+    DEBUG("Unable to parse N2kMsg with PGN %i", msg.PGN);
     return _invalidSku;
   }
 }
-
+// ****************************************************************************
+//  PGN 128267  Water depth
+// ****************************************************************************
 const SKUpdate& SKNMEA2000Parser::parse128267(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
   unsigned char sid;
-  double depthBelowTransducer;
-  double offset;
+  double depthBelowTransducer = N2kDoubleNA;
+  double offset = N2kDoubleNA;
 
   if (ParseN2kWaterDepth(msg, sid, depthBelowTransducer, offset)) {
-    SKUpdateStatic<3> *update = new SKUpdateStatic<3>();
-    update->setTimestamp(timestamp);
+    if (!N2kIsNA(depthBelowTransducer)) {
+      SKUpdateStatic<3> *update = new SKUpdateStatic<3>();
+      update->setTimestamp(timestamp);
 
-    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
-    update->setSource(source);
+      SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+      update->setSource(source);
 
-    update->setEnvironmentDepthBelowTransducer(depthBelowTransducer);
+      update->setEnvironmentDepthBelowTransducer(depthBelowTransducer);
 
-    if (!N2kIsNA(offset)) {
```

Comment:
I was wrong when I told you we did not need this anymore. Removing this broke the tests. I am re-adding it.